src/main/java/com/google/sps/servlets/GroceryServlet.java

```
@@ -0,0 +1,130 @@
+public class GroceryServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {      
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    // if user isn't associated with family don't allow them to create a list
+    Query query = new Query("UserInfo")
+        .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
+    PreparedQuery results = datastore.prepare(query);
+    Entity userInfoEntity = results.asSingleEntity();
+    if (userInfoEntity == null) {
+        response.sendRedirect("/grocery.html");
+        return;
+    }
+    String assignGrocery = request.getParameter("assignGrocery");
+    String noneAssigned = " ";
+    Entity groceryEntity = new Entity("Grocery");
+   
+    // turn familyID into a key so the users family can be accessed
+    long familyID = (long) userInfoEntity.getProperty("familyID");
+    Key familyEntityKey = KeyFactory.createKey("Family", familyID);
+    Entity familyEntity;
+    try {
+        familyEntity = datastore.get(familyEntityKey);
+    } catch (EntityNotFoundException e) {
+        System.out.println("Family not found");
+        return;
+    }
+
+    // checks if the given email matches a email in the family
+    ArrayList<String> memberEmails = (ArrayList<String>) familyEntity.getProperty("memberEmails");
+    for(int i = 0; i < memberEmails.size(); i++) {
+        System.out.println("Cuerrnt email is : " + memberEmails.get(i));
+        if(assignGrocery.equals(memberEmails.get(i))) {
+            groceryEntity.setProperty("assignEmail" , memberEmails.get(i));
+            System.out.println("emails match");
+            break;
+        } else {
+            groceryEntity.setProperty("assignEmail", noneAssigned);
+        }
```

I think you'll need to also consider what happens if `assignGrocery` is provided but that person isn't part of the family. Should you fail the request?

---

I want to make it return an error to the user. I would have to look up on error handling so this may take awhile

---

well actually would you prefer I work error handling and try to return that or just not add it to the list. I ask this because if it was an error I would want to return a error instead of a JSON and jenny said that's more work than.

---

Ok so I just made it to where nothing is added to the datastore if that problem occurs, is that fine?

---

I think the correct approach would be to throw an error.